File: 804-unique-morse-code-words.py
class Solution:
    arr = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
    def uniqueMorseRepresentations(self, words: List[str]) -> int:
        transform = set()
        for word in words:
            form = "".join(self.arr[ord(ch) - ord('a')] for ch in word)
            transform.add(form)
        return len(transform)

File: 804-unique-morse-code-words.java
class Solution {
    String[] arr = {".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
    public int uniqueMorseRepresentations(String[] words) {
        Set<String> Transf = new HashSet<>();
        for(String word : words){
            String form = "";
            for(Character ch : word.toCharArray()){
                form += arr[(int)ch-(int)'a'];
            }
            Transf.add(form);
        }
        return Transf.size();
    }
}
